spec/defines/openssl_certificate_x509_spec.rb

```diff
@@ -14,9 +16,9 @@
     end
 
     it 'fails' do
-      expect {
+      expect do
         is_expected.to contain_file('/etc/ssl/certs/foo.cnf')
-      }.to raise_error(%r{got ['barz'|String]})
+      end.to raise_error(%r{got ['barz|Sting]})
```

Comment:
This (and others in this file) should really be changed to:
is_expected.to compile.and_raise_error(%r{got ['barz|Sting]})

Reply:
should we remove the `contain_file` test or do you want me to keep them?

```diff
diff --git a/spec/defines/openssl_export_pem_cert_spec.rb b/spec/defines/openssl_export_pem_cert_spec.rb
index a296c7f..5bb4b30 100644
--- a/spec/defines/openssl_export_pem_cert_spec.rb
+++ b/spec/defines/openssl_export_pem_cert_spec.rb
@@ -17,11 +17,7 @@ describe 'openssl::export::pem_cert' do
       }
     end

-    it 'fails' do
-      expect do
-        is_expected.to contain_exec('Export /etc/ssl/certs/foo.pfx to /etc/ssl/certs/foo.pem')
-      end.to raise_error(Puppet::Error, %r{either})
-    end
+    it { is_expected.to compile.and_raise_error(%r{either}) }
   end

   context 'when enable and pfx and der cert is provided' do
```

Reply:
I can take a look at changing those, but would prefer to do it after this PR got merged